### PR TITLE
Update DG and added some UI checks when changing mode

### PR DESCRIPTION
**Pull request by farice9**
- Made gitignore to ignore the entire data and logs folder
- Middle of standardizing UI
- Didn't checkout new branch before updating DG hence here I am

If it fails ShowLinkParserTest I think we can still merge this first as my parts should not affect Yuheng's part

### CODE REVIEW

**Comment by lingsihui** [comment]
Hi please correct checkstyle error !

**Comment by farice9** [comment]
Fixed! Sorry I missed it, thanks for pointing it out!

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -13,18 +49,103 @@ If you plan to use Intellij IDEA (highly recommended):
...
+* **Flashcard**
+
+Each of these modes has their own unique commands. The user must switch to that mode to perform the
+mode's functionalities.
+
+User can switch between these modes at any point when using the software.
+
+**How the architecture components interact with each other**
+
+![Sequence Overview](Images/ArchitectureDiagrams/overviewsequence.png)
```
Comment (luziyi9898) [inline docs/DeveloperGuide.md:117]: LGTM!

File: src/main/java/userinterface/HelpMessage.java [hunk 2]
```diff
@@ -19,25 +19,30 @@
...
-            + "su grade <index number>                   - sus grade at specified index\n";
+            + "delete grade INDEX_NUMBER                 - deletes grade at specified index\n"
+            + "su grade INDEX_NUMBER                     - SU the grade at specified index\n"
+            + "star grade INDEX_NUMBER                   - marks the grade as star\n";
 
     private static final String bookmarkCommands = "Here are the bookmark commands available:\n"
-            + "bm <category index>  - changes mode from bookmark main into a category \n"
-            + "add <link>           - add bookmark link into a specific category\n"
-            + "rm <index number>    - remove a bookmark link into a specific category\n"
+            + "bm CATEGORY_INDEX    - changes mode from bookmark main into a category \n"
```
Comment (luziyi9898) [inline src/main/java/userinterface/HelpMessage.java:37]: Thanks for standardising.

**Review by luziyi9898 — COMMENTED** [review]
LGTM

**Pull request merged**